### ISSUE TITLE
Okiba Reckoner Raid second lore quits the game

### ISSUE THREAD

**Issue by Jipsu** [magefree/mage#9331]
[[Okiba Reckoner Raid]] causes a player to lose the game because adding lore counter 2 gets stuck in a loop: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer

![image](https://user-images.githubusercontent.com/22531697/181624606-b51a13ee-b0d5-4612-a125-99561b17a880.png)

10:16 PM: Turn 10 smolpp (20 - 21)
10:16 PM, T10.DR: smolpp draws a card
10:17 PM, T10.M1: smolpp casts Okiba Reckoner Raid [d89]
10:17 PM, T10.M1: smolpp puts Okiba Reckoner Raid [d89] from stack onto the Battlefield
10:17 PM, T10.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - I, II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T10.M1: Jipsu loses 1 life from Okiba Reckoner Raid [d89]
10:17 PM, T10.M1: smolpp gains 1 life (source: Okiba Reckoner Raid [d89])
10:17 PM, T10.DA: smolpp skip attack
10:17 PM: Turn 11 Jipsu (21 - 20)
10:17 PM, T11.DR: Jipsu draws a card
10:17 PM, T11.DA: Jipsu attacks with 2 creatures
10:17 PM, T11.DB: Attacker: Bear Token [fdb] (2/2) blocked by Ravenous Rats [5a0] (1/1)
10:17 PM, T11.DB: Attacker: Bear Token [8ad] (2/2) unblocked
10:17 PM, T11.CD: Bear Token [fdb] deals 2 damage to Ravenous Rats [5a0]
10:17 PM, T11.CD: Ravenous Rats [5a0] deals 1 damage to Bear Token [fdb]
10:17 PM, T11.CD: smolpp loses 2 life at combat from Bear Token [8ad]
10:17 PM, T11.CD: Ravenous Rats [5a0] died
10:17 PM, T11.M2: Jipsu casts Mist-Syndicate Naga [1e5]
10:17 PM, T11.M2: Jipsu puts Mist-Syndicate Naga [1e5] from stack onto the Battlefield
10:17 PM: Turn 12 smolpp (19 - 20)
10:17 PM, T12.DR: smolpp draws a card
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: Auto-restored to T12.M1 due game error: mage.player.human.HumanPlayer cannot be cast to mage.filter.predicate.ObjectSourcePlayer
10:17 PM, T12.M1: smolpp - Ability triggers: Okiba Reckoner Raid [d89] - II - Each opponent loses 1 life and you gain 1 life.
10:17 PM, T12.M1: smolpp has won the game
10:17 PM, T12.M1: Jipsu has lost the game.

**Comment by awjackson** [comment]
I can't reproduce this and Okiba Reckoner Raid is an extremely simple Saga card with no custom code. The error must have been caused by some other card.

**Comment by awjackson** [comment]
This looks like a duplicate of https://github.com/magefree/mage/issues/8086

**Comment by awjackson** [comment]
I think the error might have actually been caused by the card smolpp drew at T12.DR. Is there any possibility of identifying that card from the server log?

**Comment by Alex-Vasile** [comment]
There are like 100+ cards which use `ObjectSourcePlayer`. 

@Jipsu what format was this in and what colors were people playing? Maybe that can help narrow this down.
It looks like you were playing Blue and possibly Green, and smolpp was playing Black?
